pkg/ext/delegate_error_test.go

```diff
@@ -0,0 +1,1109 @@
+				completed:    false,
+				err:          errMissingUserInfo,
+			},
+			wantErr: true,
+		},
+		{
+			name: "store get error",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {
+				ms.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, assert.AnError)
+			},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				name:    "test-object",
+				options: &metav1.DeleteOptions{},
+			},
+			expect: output{
+				deleteResult: nil,
+				completed:    false,
+				err:          assert.AnError,
+			},
+			wantErr: true,
+		},
+		{
+			name: "store delete error",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {
+				ms.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(&TestType{}, nil)
+				ms.EXPECT().Delete(gomock.Any(), gomock.Any(), gomock.Any()).Return(assert.AnError)
+			},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				name:    "test-object",
+				options: &metav1.DeleteOptions{},
+			},
+			expect: output{
+				deleteResult: &TestType{},
+				completed:    true,
+				err:          assert.AnError,
+			},
+			wantErr: true,
+		},
+		{
+			name: "delete validation error",
+			storeSetup: func(ms *MockStore[*TestType, *TestTypeList]) {
+				ms.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(&TestType{}, nil)
+			},
+			input: input{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				name:             "test-object",
+				deleteValidation: func(ctx context.Context, obj runtime.Object) error { return assert.AnError },
+				options:          &metav1.DeleteOptions{},
+			},
+			expect: output{
+				deleteResult: nil,
+				completed:    false,
+				err:          assert.AnError,
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			addToSchemeTest(scheme)
+
+			gvk := schema.GroupVersionKind{Group: "ext.cattle.io", Version: "v1", Kind: "TestType"}
+			gvr := schema.GroupVersionResource{Group: "ext.cattle.io", Version: "v1", Resource: "testtypes"}
+
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := NewMockStore[*TestType, *TestTypeList](ctrl)
+			tt.storeSetup(mockStore)
+
+			testDelegate := &delegateError[*TestType, *TestTypeList]{
+				inner: &delegate[*TestType, *TestTypeList]{
+					scheme: scheme,
+					t:      &TestType{},
+					tList:  &TestTypeList{},
+					gvk:    gvk,
+					gvr:    gvr,
+					store:  mockStore,
+				},
+			}
+
+			result, completed, err := testDelegate.Delete(tt.input.ctx, tt.input.name, tt.input.deleteValidation, tt.input.options)
+			if tt.wantErr {
+				// check if we have an error
+				assert.Error(t, err)
+
+				// check if error is an apierror
+				if _, ok := err.(apierrors.APIStatus); ok {
+					assert.True(t, ok)
+					return
+				}
```

Comment:
Should this be

```suggestion
				_, ok := err.(apierrors.APIStatus)
				assert.True(t, ok)
```

? Right now if the error returned is not an APIStatus there just won't be any assert.